$NetBSD: patch-ab,v 1.2 1998/06/04 13:08:36 agc Exp $

Note: I (tsarna@endicor.com) have submitted the patches in this file 
plus a more generic Makefile patch (ie, without the pkgsrc changes) to
the Samba group. I hope this file can be deleted when this package is
updated for the next version of Samba. 

--- includes.h.orig	Tue May 12 20:46:49 1998
+++ includes.h	Tue Jun  2 15:46:07 1998
@@ -198,7 +198,7 @@
 #include <pwdadj.h>
 #endif
 
-#if defined(SHADOW_PWD) && !defined(NETBSD) && !defined(FreeBSD) && !defined(CONVEX) && !defined(__OpenBSD__)
+#if defined(SHADOW_PWD) && !defined(CONVEX)
 #include <shadow.h>
 #endif
 
@@ -506,21 +506,22 @@
 #endif
 
 
-#ifdef NETBSD 
-#ifdef NetBSD1_3
-#include <string.h>
+
+#ifdef __NetBSD__
+#include <dirent.h>
+#include <strings.h>
+#include <netinet/tcp.h>
 #ifdef ALLOW_CHANGE_PASSWORD
 #include <termios.h>
 #endif /* ALLOW_CHANGE_PASSWORD */
-#else /* NetBSD1_3 */
-#include <strings.h>
-#endif /* NetBSD1_3 */
-#include <netinet/tcp.h>
-/* you may not need this */
-#define NO_GETSPNAM
 #define SIGNAL_CAST (void (*)())
-#define USE_DIRECT
-#define REPLACE_INNETGR
+#define HAVE_BZERO
+#define HAVE_PATHCONF
+#define HAVE_GETGRNAM 1
+#define HAVE_GETTIMEOFDAY
+#define HAVE_MEMMOVE
+#define USE_GETCWD
+#define USE_SETSID
 #endif 
 
 

--- interface.c.orig	Fri Jan  9 17:22:01 1998
+++ interface.c	Fri Jan  9 17:22:13 1998
@@ -161,7 +161,7 @@
       }
     }
   }
-#elif defined(__FreeBSD__) || defined(NETBSD) || defined(AMIGA) || defined(_AIX41) || defined(__OpenBSD__)
+#elif defined(__FreeBSD__) || defined(__NetBSD__) || defined(AMIGA) || defined(_AIX41) || defined(__OpenBSD__)
   ifc.ifc_len = sizeof(buff);
   ifc.ifc_buf = buff;
   if (ioctl(sock, SIOCGIFCONF, &ifc) < 0) {
--- quotas.c.orig	Fri Jan  9 17:27:17 1998
+++ quotas.c	Fri Jan  9 17:29:13 1998
@@ -546,7 +546,7 @@
 
 #else
 
-#if    defined(__FreeBSD__) || defined(__OpenBSD__)
+#if    defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
 #include <ufs/ufs/quota.h>
 #include <machine/param.h>
 #elif         AIX
@@ -556,7 +556,7 @@
 #define dqb_curfiles dqb_curinodes
 #define dqb_fhardlimit dqb_ihardlimit
 #define dqb_fsoftlimit dqb_isoftlimit
-#else /* !__FreeBSD__ && !AIX && !__OpenBSD__ */
+#else /* !__FreeBSD__ && !AIX && !__OpenBSD__ && !__NetBSD__ */
 #include <sys/quota.h>
 #include <devnm.h>
 #endif
@@ -569,7 +569,7 @@
   uid_t euser_id;
   int r;
   struct dqblk D;
-#if !defined(__FreeBSD__) && !defined(AIX) && !defined(__OpenBSD__)
+#if !defined(__FreeBSD__) && !defined(AIX) && !defined(__OpenBSD__) && !defined(__NetBSD__)
   char dev_disk[256];
   struct stat S;
   /* find the block device file */
@@ -591,7 +591,7 @@
       DEBUG(5,("Unable to reset uid to %d\n", user_id));
   }
 #else /* USE_SETRES */
-#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
   {
     /* FreeBSD patches from Marty Moll <martym@arbor.edu> */
     uid_t user_id;
@@ -615,17 +615,17 @@
   /* AIX has both USER and GROUP quotas: 
      Get the USER quota (ohnielse@fysik.dtu.dk) */
   r= quotactl(path,QCMD(Q_GETQUOTA,USRQUOTA),euser_id,(char *) &D);
-#else /* !__FreeBSD__ && !AIX && !__OpenBSD__ */
+#else /* !__FreeBSD__ && !AIX && !__OpenBSD__ && !__NetBSD__ */
   r=quotactl(Q_GETQUOTA, dev_disk, euser_id, &D);
-#endif /* !__FreeBSD__ && !AIX && !__OpenBSD__ */
+#endif /* !__FreeBSD__ && !AIX && !__OpenBSD__ && !__NetBSD__ */
 #endif /* USE_SETRES */
 
   /* Use softlimit to determine disk space, except when it has been exceeded */
-#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
   *bsize = DEV_BSIZE;
-#else /* !__FreeBSD__ && !__OpenBSD__ */
+#else /* !__FreeBSD__ && !__OpenBSD__ && !__NetBSD__ */
   *bsize = 1024;
-#endif /*!__FreeBSD__ && !__OpenBSD__ */
+#endif /*!__FreeBSD__ && !__OpenBSD__ && !__NetBSD__ */
 
   if (r)
     {
@@ -641,7 +641,7 @@
     return(False);
   /* Use softlimit to determine disk space, except when it has been exceeded */
   if ((D.dqb_curblocks>D.dqb_bsoftlimit)
-#if !defined(__FreeBSD__) && !defined(__OpenBSD__)
+#if !defined(__FreeBSD__) && !defined(__OpenBSD__) && !defined(__NetBSD__)
 ||((D.dqb_curfiles>D.dqb_fsoftlimit) && (D.dqb_fsoftlimit != 0))
 #endif
     ) {
--- util.c.orig	Fri Jan  9 17:22:58 1998
+++ util.c	Fri Jan  9 17:23:05 1998
@@ -3352,7 +3352,7 @@
   
   bzero((char *)&sock,sizeof(sock));
   memcpy((char *)&sock.sin_addr,(char *)hp->h_addr, hp->h_length);
-#if defined(__FreeBSD__) || defined(NETBSD) || defined(__OpenBSD__) /* XXX not the right ifdef */
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) /* XXX not the right ifdef */
   sock.sin_len = sizeof(sock);
 #endif
   sock.sin_port = htons( port );
